packages/graphql-language-service-utils/src/validateWithCustomRules.ts

```diff
@@ -7,12 +7,19 @@
  *
  */
 
-import { DocumentNode, TypeDefinitionNode } from 'graphql/language';
-import { GraphQLError } from 'graphql/error';
-import { GraphQLSchema } from 'graphql/type';
 import { CustomValidationRule } from 'graphql-language-service-types';
 
-import { specifiedRules, TypeInfo, validate } from 'graphql';
+import {
+  DocumentNode,
+  Kind,
+  specifiedRules,
+  validate,
+  GraphQLError,
+  GraphQLSchema,
+  NoUnusedFragmentsRule,
+  KnownFragmentNamesRule,
+} from 'graphql';
+import { ExecutableDefinitions } from 'graphql/validation/rules/ExecutableDefinitions';
```

Comment:
Here are the main changes, rest is just refactoring.

Reply:
awesome, thank you @IvanGoncharov ! this will come in very handy